pkg/compare/compare.go

```diff
@@ -90,11 +90,56 @@ func YAMLCmpWithIgnore(a, b string, ignorePaths []string) string {
 	if err := yaml.Unmarshal([]byte(b), &bo); err != nil {
 		return err.Error()
 	}
+
+	unmarshalConfigMapData(ao)
+	unmarshalConfigMapData(bo)
+
 	var r YAMLCmpReporter
 	cmp.Equal(ao, bo, cmp.Reporter(&r), genPathIgnoreOpt(ignorePaths))
 	return r.String()
 }
 
+func unmarshalConfigMapData(obj map[string]interface{}) {
+	InlineYamlData(obj, "ConfigMap", "data")
+}
+
+func InlineYamlData(obj map[string]interface{}, targetKind, targetPath string) {
+	if kind, ok := obj["kind"]; !ok || kind != targetKind {
+		return
```

Comment:
suggest to add some error message for unexpected return.

Reply:
Move this "kind check" out of this func to decouple the responsibilities.

Also added errors and logs for other unexpected scenarios.